src/wxflow/scheduler/scheduler.py

```diff
@@ -0,0 +1,189 @@
+from datetime import timedelta
+from typing import Any, Dict, List, Optional, Union
+
+from ..attrdict import AttrDict
+from ..factory import Factory
+from ..timetools import timedelta_to_HMS, to_timedelta
+
+
+class Scheduler:
+
+    scheduler_factory = Factory('Scheduler')
+
+    def __init__(self, config: Dict[str, Any], *args: Any, **kwargs: Any) -> None:
+        """
+        Initializes the scheduler with the provided configuration.
+
+        Parameters
+        ----------
+        config : Dict
+            Configuration dictionary for the scheduler.
+        *args : Any
+            Additional positional arguments.
+        **kwargs : Any
+            Additional keyword arguments.
+
+        """
+
+        # Cache incoming config
+        self._config = AttrDict(config)
+
+        self.specs = self._config_to_specs
```

Comment:
This would then become

```suggestion
        self._config_to_specs()
```